[Gra7ityIC3] iP

Gra7ityIC3 opened the pull request:
### MorganaPro
> “I am NOT a cat... Yeah, I guess I am.” – Morgana ([source](https://youtu.be/Ih5pEDV0T9M?si=FNjprolF2GPrtYMe&t=937))

MorganaPro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is,

1. download it from [here](https://github.com/Gra7ityIC3/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :smile_cat:

And it is **FREE**!

Features:

- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders (coming soon)

---

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Gra7ityIC3 committed:
Rename main class and add greeting

Gra7ityIC3 committed:
Add support for echoing user input

Gra7ityIC3 committed:
Add support for listing user input

Gra7ityIC3 committed:
Add support for marking tasks as done/not done

Gra7ityIC3 committed:
Add support for tracking three types of tasks - todos, deadlines, and events

Gra7ityIC3 committed:
Add automated text UI testing

Gra7ityIC3 committed:
Add missing colon in output of event task

Gra7ityIC3 committed:
Handle errors when creating new task

Gra7ityIC3 committed:
Add support for deleting tasks

Gra7ityIC3 committed:
Add enum for task types

Gra7ityIC3 committed:
Handle errors when marking/unmarking/deleting tasks

Gra7ityIC3 committed:
Add support for saving tasks to the hard disk

Gra7ityIC3 committed:
Merge branch 'branch-Level-7'

Gra7ityIC3 committed:
Change format of date/time from String to LocalDateTime

Gra7ityIC3 committed:
Merge branch 'branch-Level-8'

Gra7ityIC3 committed:
Refactor code to enhance object-oriented design

Extracted the following classes:
- Ui: Handles interactions with the user
- Storage: Manages loading and saving tasks to a file
- Parser: Interprets and parses user commands
- TaskList: Encapsulates the task list and its operations

Gra7ityIC3 committed:
Merge branch 'A-MoreOOP'

Gra7ityIC3 committed:
Organize classes into suitable java packages

Gra7ityIC3 committed:
Merge branch 'A-Packages'

Gra7ityIC3 committed:
Merge branch 'add-gradle-support'

Gra7ityIC3 committed:
Modify build.gradle

Gra7ityIC3 committed:
Add JUnit tests for DeadlineCommand and EventCommand

Gra7ityIC3 committed:
Merge branch 'branch-A-JUnit'

Gra7ityIC3 committed:
Add JavaDoc comments to Task, TaskList, and Storage classes

Gra7ityIC3 committed:
Update code to follow the coding standard

Gra7ityIC3 committed:
Add support for finding tasks by a keyword

Gra7ityIC3 committed:
Merge branch 'branch-A-JavaDoc'

Gra7ityIC3 committed:
Merge branch 'branch-A-CodingStandard'

Gra7ityIC3 committed:
Merge branch 'branch-Level-9'

Gra7ityIC3 committed:
Add Checkstyle support and Javadoc comments

Gra7ityIC3 committed:
Merge branch 'branch-A-CheckStyle'

HanB1n commented on src/main/java/morgana/task/TaskList.java:
```diff
@@ -0,0 +1,64 @@
+ * Represents a list of tasks, providing methods to add, remove, and
+ * retrieve tasks from the list.
+ */
+public class TaskList {
```
Would it be better if the method names were more descriptive eg. addTask, getTask, removeTask, instead of simply add, get, remove?

HanB1n commented on src/main/java/morgana/exceptions/MorganaException.java:
```diff
@@ -0,0 +1,17 @@
+     *
+     * @param message The detail message describing the specific error that occurred.
+     */
+    public MorganaException(String message) {
```
Would the code be more readable if different error types are extracted?

Gra7ityIC3 replied:
You're right. Thank you for the suggestion.

HanB1n reviewed (commented):
The code was well written and easy to understand with insignificant coding standard violations. Good Job!

leroytan commented on src/main/java/morgana/Morgana.java:
```diff
@@ -0,0 +1,63 @@
+ * Initializes the application and starts the interaction with the user.
+ */
+public class Morgana {
+    private static final String NAME = "Morgana";
```
I like how you used caps for a constant

leroytan commented on src/main/java/morgana/commands/AddCommand.java:
```diff
@@ -0,0 +1,46 @@
+ * to define how the task is created based on the user's input.
+ */
+public abstract class AddCommand extends Command {
+    private final String args;
```
Should this be in caps instead since it is a final variable?

Gra7ityIC3 replied:
`args` is not a constant so it does not need to be capitalized.

leroytan commented on src/main/java/morgana/commands/FindCommand.java:
```diff
@@ -0,0 +1,38 @@
+ * Represents a command to find tasks that match a given keyword in their description.
+ */
+public class FindCommand extends Command {
+    private final String keyword;
```
Consider changing the keyword to CAPS, I have also seen the issue elsewhere in the code

leroytan commented on src/test/java/morgana/DeadlineCommandTest.java:
```diff
@@ -0,0 +1,56 @@
+
+public class DeadlineCommandTest {
+    @Test
+    public void deadlineCommand_validInput_success() throws MorganaException {
```
I like how u named the testcases

leroytan reviewed (commented):
LGTM, just need to change some final variable to CAPS

Gra7ityIC3 committed:
Add GUI to chatbot using JavaFX

Gra7ityIC3 committed:
Merge branch 'branch-Level-10'

Gra7ityIC3 committed:
Fix JUnit tests and remove text-based UI

Gra7ityIC3 committed:
Add Java assertions when parsing tasks from strings

Gra7ityIC3 committed:
Merge pull request #1 from Gra7ityIC3/branch-A-Assertions

Add Java assertions when parsing tasks from strings

Gra7ityIC3 committed:
Improve code quality

The command classes were previously handling task-related logic,
such as listing, finding, and marking tasks, which leads to code
duplication and reduced maintainability.

Let's delegate these responsibilities to the TaskList class to
centralize task management, simplify the command classes, and
improve separation of concerns, resulting in better modularity.

Additionally, the DateTimeUtil class was introduced to define common
date and time formatting patterns and messages, ensuring consistency
and reusability across the application. Magic literals were also
refactored to constants to improve readability.

Gra7ityIC3 committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/morgana/parser/Parser.java

Gra7ityIC3 committed:
Merge pull request #2 from Gra7ityIC3/branch-A-CodeQuality

Improve code quality

Gra7ityIC3 committed:
Set up CI with GitHub Actions

Gra7ityIC3 committed:
Detect scheduling anomalies

Previously, events could overlap, leading to scheduling conflicts.

Overlapping events reduce the clarity of the schedule and make task
management confusing.

Let's,
* add a method to detect overlapping events
* throw an exception if a new event clashes with another event

Gra7ityIC3 committed:
Merge branch 'branch-B-DetectAnomalies'

Gra7ityIC3 committed:
Add more JUnit tests

Gra7ityIC3 committed:
Merge branch 'branch-A-MoreTesting'

Gra7ityIC3 committed:
Update user guide

Gra7ityIC3 committed:
Update Ui.png

Gra7ityIC3 committed:
Update user guide

Gra7ityIC3 committed:
Update user guide

Gra7ityIC3 committed:
Update user guide

Gra7ityIC3 committed:
Fix phrasing: 'be created' -> 'can be created'
